### PR TITLE
[luci-interpreter] Support Integer Less

### PR DESCRIPTION
**Pull request by binarman**
This PR enables Less operation for S32 and S64.

Related issue: #8155 

ONE-DCO-1.0-Signed-off-by: Alexander Efimov <a.efimov@samsung.com>

### CODE REVIEW

File: compiler/luci-interpreter/src/kernels/Less.test.cpp [hunk 1]
```diff
@@ -223,6 +299,36 @@ TEST_F(LessTest, Input_Output_Type_NEG)
...
+}
+
+TEST_F(LessTest, Int32_Broadcast_NEG)
+{
+  Tensor x_tensor = makeInputTensor<DataType::S32>({2}, {1, 2}, _memory_manager.get());
+  Tensor y_tensor = makeInputTensor<DataType::S32>({3}, {1, 2, 3}, _memory_manager.get());
+  Tensor output_tensor = makeOutputTensor(DataType::BOOL);
+
+  Less kernel(&x_tensor, &y_tensor, &output_tensor);
+  ASSERT_DEATH(kernel.configure(), ".*");
```
Comment (binarman) [inline compiler/luci-interpreter/src/kernels/Less.test.cpp:319]:
```suggestion
  ASSERT_ANY_THROW(kernel.configure());
```

File: compiler/luci-interpreter/src/kernels/Less.test.cpp [hunk 2]
```diff
@@ -223,6 +299,36 @@ TEST_F(LessTest, Input_Output_Type_NEG)
...
 }
 
+TEST_F(LessTest, Float_Broadcast_NEG)
+{
+  Tensor x_tensor = makeInputTensor<DataType::FLOAT32>({2}, {1.f, 2.f}, _memory_manager.get());
+  Tensor y_tensor = makeInputTensor<DataType::FLOAT32>({3}, {1.f, 2.f, 3.f}, _memory_manager.get());
+  Tensor output_tensor = makeOutputTensor(DataType::BOOL);
+
+  Less kernel(&x_tensor, &y_tensor, &output_tensor);
+  ASSERT_DEATH(kernel.configure(), ".*");
```
Comment (binarman) [inline compiler/luci-interpreter/src/kernels/Less.test.cpp:309]:
```suggestion
  ASSERT_ANY_THROW(kernel.configure());
```

File: compiler/luci-interpreter/src/kernels/Less.test.cpp [hunk 3]
```diff
@@ -223,6 +299,36 @@ TEST_F(LessTest, Input_Output_Type_NEG)
...
+}
+
+TEST_F(LessTest, Int64_Broadcast_NEG)
+{
+  Tensor x_tensor = makeInputTensor<DataType::S64>({2}, {1, 2}, _memory_manager.get());
+  Tensor y_tensor = makeInputTensor<DataType::S64>({3}, {1, 2, 3}, _memory_manager.get());
+  Tensor output_tensor = makeOutputTensor(DataType::BOOL);
+
+  Less kernel(&x_tensor, &y_tensor, &output_tensor);
+  ASSERT_DEATH(kernel.configure(), ".*");
```
Comment (binarman) [inline compiler/luci-interpreter/src/kernels/Less.test.cpp:329]:
```suggestion
  ASSERT_ANY_THROW(kernel.configure());
```

File: compiler/luci-interpreter/src/kernels/Utils.cpp [hunk 4]
```diff
@@ -175,7 +175,11 @@ Shape calculateShapeForBroadcast(const Shape &input1_shape, const Shape &input2_
   {
     const int32_t input1_dim = i < num_input1_dims ? input1_shape.dim(num_input1_dims - i - 1) : 1;
     const int32_t input2_dim = i < num_input2_dims ? input2_shape.dim(num_input2_dims - i - 1) : 1;
-    assert(input1_dim == input2_dim || input1_dim == 1 || input2_dim == 1);
+
+    bool need_broadcast = input1_dim != input2_dim;
+    bool can_broadcast = input1_dim == 1 || input2_dim == 1;
+    LUCI_INTERPRETER_CHECK(!need_broadcast || can_broadcast);
+
```
Comment (binarman) [inline compiler/luci-interpreter/src/kernels/Utils.cpp:182]: This part already merged, it is present here just to enable tests without rebase

**Review by m-bronnikov — APPROVED** [review]
LGTM

**Pull request merged**